server/service/apple_mdm.go

```diff
@@ -3983,6 +4029,96 @@ func preprocessProfileContents(
 	return nil
 }
 
+type digiCertVarsFound struct {
+	data     bool
+	password bool
+}
+
+// Ok makes sure that both DATA and PASSWORD variables are present in a DigiCert profile.
+func (d digiCertVarsFound) Ok() bool {
+	return d.data && d.password || !d.data && !d.password
+}
+
+func getIsDigiCertConfiguredFunc(ctx context.Context, appConfig *fleet.AppConfig, ds fleet.Datastore,
```

Comment:
That's a bit unexpected as-is as it's only used in one place (that I can see), not sure why we call a func to create another func for this, is that a design for a longer-term plan? Otherwise the inner func could be inside the func that calls it so it wouldn't need the double-call.